src/datasets/arrow_dataset.py

```diff
@@ -223,31 +196,23 @@ def wrapper(*args, **kwargs):
 
 
 class Dataset(DatasetInfoMixin, IndexableMixin):
-    """A Dataset backed by an Arrow table or Record Batch."""
+    """A Dataset backed by an Arrow table."""
 
     def __init__(
         self,
-        arrow_table: pa.Table,
-        data_files: Optional[List[dict]] = None,
+        arrow_table: Union[pa.Table, Table],
```

Comment:
same question, why not rely only on your new abstraction?

Reply:
I liked the idea of allowing users to use regular arrow tables as well, but maybe it's simpler if we only allow the new abstraction

Reply:
Yes I think it would make everything clearer, documentation wise - since the typing here is not enforced.
Contractually (e.g. not statically but for runtime) a `pa.Table` is a `Table` too, so this use is allowed too? not sure this is a very good point to make, but whenever we can simplify stuff... :)

Reply:
Yes this is allowed since they have the exact same API

Reply:
yes, my point was that it would mean that our documentation-typing is slightly misleading, but since we don't have interfaces, what can we do, right? 🤷 
so still in favour of the simplification.